sphinx/application/common/resources/src/main/java/chat/sphinx/resources/ContextExtensions.kt

```diff
@@ -17,7 +18,36 @@ inline val Context.inputMethodManager: InputMethodManager?
 @ColorInt
 @Suppress("NOTHING_TO_INLINE")
 inline fun Context.getRandomColor(): Int {
-    val randomColorResource = listOf<@ColorRes Int>(
+    return ContextCompat.getColor(this, getRandomColorRes())
+}
+
+@ColorInt
+@Suppress("NOTHING_TO_INLINE")
+inline fun Context.getColorForKey(colorKey: String): Int {
+
+    this.getSharedPreferences("sphinx_colors", Context.MODE_PRIVATE).let { sharedPrefs ->
+        val colorHexCode = sharedPrefs.getString(colorKey, null)?.let { hexCode ->
```

Comment:
Querying SharedPreferences needs to happen on a background thread as it is Disk IO.